omc3/madx_wrapper.py

```diff
@@ -186,12 +186,20 @@ def _raise_madx_error(log=None, file=None):
     message = "MADX run failed."
     if log is not None:
         try:
-            with open(log, "r") as f:
-                content = f.readlines()
-            if content[-1].startswith("+="):
-                message += f" '{content[-1].replace('+=+=+=', '').strip()}'."
-        except (IOError, IndexError):
+            content = Path(log).read_text()
+        except IOError:
             pass
+        else:
+            if "warning: Twiss failed:  MAD-X continues" in content:
+                message += " At least one twiss failed."
+
+            try: 
+                last_line = content.splitlines()[-1]
+            except IndexError:
+                pass
+            else:
+                if last_line.startswith("+="):
+                    message += f" '{last_line.replace('+=+=+=', '').strip()}'."
```

Comment:
Nice addition. I guess it couldn't hurt to have a comment explaining the logic of this last line.

Reply:
Or did you mean the "if +=+= is in the line, that means MAD-X wrote an error message there?"

Reply:
Yeah something like this. I kinda forgot about the MAD-X kinks with these errors.